projects/core/src/lib/map/map-layer.component.ts

```diff
@@ -311,10 +311,18 @@ export class OlRasterLayerComponent
 
         this.source.setTileLoadFunction((tile, src) => {
             const client = new XMLHttpRequest();
+
+            const sub = this.projectService.getQueryAbortStream().subscribe((_) => {
+                client.abort();
+                sub.unsubscribe();
+                this.source.dispatchEvent('tileloaderror');
+            });
+
```

Comment:
Could there be a race condition in doing this?

Reply:
I think it could happen that the request completes exactly at the same time as we try to abort it. But `abort` does nothing if the request already finished. It could happen that the `tileloadend` and `tileloaderror` both get dispatched. But as the condition for changing the rasterLayerStatus is `<=` we should be fine also. It could happen that we try to unsubscribe the subscription twice, I don't know what happens then, do you?